e2e/utils.go

```diff
@@ -82,6 +82,156 @@ var (
 	clusterID        string
 )
 
+type cephfsFilesystem struct {
+	Name           string   `json:"name"`
+	MetadataPool   string   `json:"metadata_pool"`
+	MetadataPoolID int      `json:"metadata_pool_id"`
+	DataPoolIDs    []int    `json:"data_pool_ids"`
+	DataPools      []string `json:"data_pools"`
+}
+
+// listCephFSFileSystems list CephFS filesystems in json format.
+func listCephFSFileSystems(f *framework.Framework) ([]cephfsFilesystem, error) {
+	var fsList []cephfsFilesystem
+
+	stdout, stdErr, err := execCommandInToolBoxPod(
+		f,
+		fmt.Sprintf("ceph fs ls --format=json"),
+		rookNamespace)
+	if err != nil {
+		return fsList, err
+	}
+	if stdErr != "" {
+		return fsList, fmt.Errorf("error listing fs %v", stdErr)
+	}
+
+	err = json.Unmarshal([]byte(stdout), &fsList)
+	if err != nil {
+		return fsList, err
+	}
+
+	return fsList, nil
+}
+
+// getCephFSPoolName get CephFS pool name from filesystem name.
+func getCephFSPoolName(f *framework.Framework, filesystem string) (string, error) {
+	fsList, err := listCephFSFileSystems(f)
+	if err != nil {
+		return "", err
+	}
+	for _, v := range fsList {
+		if v.Name != filesystem {
+			continue
+		}
+		return v.MetadataPool, nil
+	}
+
+	return "", os.ErrNotExist
+}
+
+// validateOmapCount validates no of OMAP entries on the given pool.
+// Works with Cephfs and RBD drivers and mode can be snaps or volumes.
+func validateOmapCount(f *framework.Framework, count int, driver, pool, mode string) {
+	var (
+		cmd    string
+		res    int
+		err    error
+		stdOut string
+		stdErr string
+	)
+	if strings.Contains("volumes", strings.ToLower(mode)) {
+		if strings.Contains("cephfs", strings.ToLower(driver)) {
+			cmd = fmt.Sprintf("rados ls --pool=%s --namespace csi | grep -v default | grep -c ^csi.volume.", pool)
+		} else if strings.Contains("rbd", strings.ToLower(driver)) {
+			cmd = fmt.Sprintf("rados ls %s | grep -v default | grep -c ^csi.volume.", rbdOptions(pool))
+		} else {
+			e2elog.Failf("unknown driver type: %s", driver)
+		}
+		stdOut, stdErr, err = execCommandInToolBoxPod(f, cmd, rookNamespace)
+		if err != nil {
+			if !strings.Contains(err.Error(), "command terminated with exit code 1") {
+				e2elog.Failf("failed to execute rados command '%s' : err=%v stdErr=%s", cmd, err, stdErr)
+			}
+		}
+		stdOut = strings.TrimSuffix(stdOut, "\n")
+		res, err = strconv.Atoi(stdOut)
+		if err != nil {
+			e2elog.Failf("failed to convert string to int: %s", stdOut)
+		}
+		if res != count {
+			e2elog.Logf("expected omap object count %d, got %d", count, res)
```

Comment:
I have intentionally used `e2elog.Logf` instead of `e2elog.Failf` in this function in a few places. I will remove them later, I just want the first run to finish without stopping on any single failure.

Reply:
do you want to move this to a **Draft** in the mean time?

Reply:
Yes, Good idea.